### ISSUE TITLE
MarkerlessAR_V2 demo will shake drastically when running

### ISSUE THREAD

**Issue by hechaochao** [ahmetozlu/Open-Source-Markerless-Augmented-Reality#1]
the register object's position is changing randomly even the camera and object keep static.Do you ever encountered this condition?

**Comment by ahmetozlu** [comment]
Yes, I have encountered with this situation many times... The main reason of it is about the redundant features. 

In this project, I am interested in corner detection (I used [ORB](https://docs.opencv.org/3.0-beta/doc/py_tutorials/py_feature2d/py_orb/py_orb.html) keypoint detector). It's done by looking for extremums of the first derivative of the image gradients in the X and Y directions. Most probably, so many corner features are detected in each camera frame and it means too large data to be processed and likely it is redundant. When the input data (feature vector) has too large data to handle, it effects the program so it works slowy. Moreover, when the input data to an algorithm is too large to be processed and it is suspected to be redundant so "AR object's position is changing randomly even the camera and object keep static" because system is confused about "How to keep positioned of the AR object? Here are so many features and most of them are redundant and changing so fastly..." then AR object starts to dance :D

**Anyway, what is the solution?** 

The solution is [Dimensionality Reduction](https://en.wikipedia.org/wiki/Dimensionality_reduction). We should get rid of the redundant data (we should eliminate feature descriptor [feature vector]) by using dimensionality reduction operations. Many data analysis software packages provide for dimension reduction such as [SciLab ](https://www.scilab.org/) and [NumPy](http://www.numpy.org/) provide some of the simpler dimensionality reduction  techniques (e.g. principal component analysis) via built-in commands. More specific algorithms are often available as publicly available scripts or third-party add-ons. Thus, we can have robust pattern detector.

Therefore, we should develop dimensionality reduction system for this project to have robust pattern detector. I am waiting your contributions because I am so busy nowadays and I can not find any time to develop this Open Source Marker-less AR Project.

**Comment by hechaochao** [comment]
@ahmetozlu Thanks for your patiently response, i'am a newbie in AR. I'm just looking for a AR project that i can use to register object in where i want it to be accurately. The rest of time i'm busy for doing my job, i'm intersted in it, once i have spare time maybe i can learn about it, then we can share.
